### DISCUSSION TITLE
Face Detection using existing database through REST API

### DISCUSSION THREAD

**Discussion by bkbilly** [immich-app/immich#12275] [Q&A] [answered]
I've added my images and immich has found all the similar people and I've added their names.
Is it possible to send an image through http post request and show the person's name?

I've exposed the immich-machine-learning port and tested it, but it doesn't return the name.
This is the request I am doing:
```python
import io
import json
import requests
from PIL import Image

img = Image.open("myimage.jpeg")
img_byte_arr = io.BytesIO()
img.save(img_byte_arr, format='PNG')
img_byte_arr = img_byte_arr.getvalue()


response = requests.post(
    "http://<myip>:3003/predict",
    data={
        "entries": json.dumps(
            {
                "facial-recognition": {
                    "detection": {"modelName": "buffalo_l", "options": {"minScore": 0.8}},
                    "recognition": {"modelName": "buffalo_l"},
                }
            }
        )
    },
    files={"image": img_byte_arr},
)
print(response.json())
```
And this is the result:
```json
{
  "facial-recognition": [
    {
      "boundingBox": {
        "x1": 285.0,
        "y1": 175.0,
        "x2": 499.0,
        "y2": 470.0
      },
      "embedding": [
        -1.0646248,
        0.8876045,
        -0.0225956,
        0.036655292,
        '........',
        -1.41123
      ],
      "score": 0.885398
    }
  ],
  "imageHeight": 800,
  "imageWidth": 800
}
```

Is there a way to get the name of the person that recognized?

**Comment by bo0tzz** [comment, marked as answer]
Immich isn't a general purpose facial recognition engine, so it doesn't expose an API like this. You'd have to match the embedding against the database directly or upload the image into Immich to for it to do the matching.

**Comment by bkbilly** [comment]
I see, thanks for the clarification!
I'll check the database that you suggested.

**Reply by JoeHogan** [reply]
@bkbilly - I'd like to do the same thing... figured immich already has curated embeddings for years worth of family and friends, would love to use that data to do instant facial recognition via API. One of my use cases is security camera's around my house to instantly recognize family and friends.

Did you ever get this working by comparing the result to the database? If so, any code you could share?
